packages/cli/test/unit/util/output/box.test.ts

```diff
@@ -0,0 +1,124 @@
+│ Hello world! │
+╰──────────────╯
+    `.trim()
+    );
+  });
+
+  it('should show single line box with padding 2', () => {
+    const result = box('Hello world!', { padding: 2 });
+    expect(stripAnsi(result)).toEqual(
+      `
+╭────────────────────────╮
+│                        │
+│                        │
+│      Hello world!      │
+│                        │
+│                        │
+╰────────────────────────╯
+    `.trim()
+    );
+  });
+
+  it('should show multiple lines with default padding', () => {
+    const result = box(
+      'Hello world!\nThis is a really, really long line of text\n\nWow!'
+    );
+    expect(stripAnsi(result)).toEqual(
+      `
+╭────────────────────────────────────────────────╮
+│                                                │
+│                  Hello world!                  │
+│   This is a really, really long line of text   │
+│                                                │
+│                      Wow!                      │
+│                                                │
+╰────────────────────────────────────────────────╯
+    `.trim()
+    );
+  });
+
+  it('should ignore ansi color escape sequences', () => {
+    const result = box(chalk.red('This text is red'));
+    expect(stripAnsi(result)).toEqual(
+      `
+╭──────────────────────╮
+│                      │
+│   This text is red   │
+│                      │
+╰──────────────────────╯
+    `.trim()
+    );
+  });
+
+  it('should left align contents', () => {
+    const result = box(
+      'This is left aligned\nThis is a really, really long line of text',
+      { textAlignment: 'left' }
+    );
+    expect(stripAnsi(result)).toEqual(
+      `
+╭────────────────────────────────────────────────╮
+│                                                │
+│   This is left aligned                         │
+│   This is a really, really long line of text   │
+│                                                │
+╰────────────────────────────────────────────────╯
+    `.trim()
+    );
+  });
+
+  it('should right align contents', () => {
+    const result = box(
+      'This is right aligned\nThis is a really, really long line of text',
+      { textAlignment: 'right' }
+    );
+    expect(stripAnsi(result)).toEqual(
+      `
+╭────────────────────────────────────────────────╮
+│                                                │
+│                        This is right aligned   │
+│   This is a really, really long line of text   │
+│                                                │
+╰────────────────────────────────────────────────╯
+    `.trim()
+    );
+  });
+
+  it('should slim if terminal width too small', () => {
+    const result = box('This is a really, really long line of text', {
+      terminalColumns: 30,
+    });
+    expect(stripAnsi(result)).toEqual(
+      `
+──────────────────────────────────────────
+
+This is a really, really long line of text
+
+──────────────────────────────────────────
+    `.trim()
+    );
+  });
```

Comment:
**requirement (blocking):** when text is longer than the number of terminal columns, the text should wrap within the narrow rules at the top and bottom

What seems to happen when the text is too long is that the top/bottom rules keep going, also wrapping:

<img width="456" alt="Screenshot 2023-02-07 at 2 17 18 PM" src="https://user-images.githubusercontent.com/41545/217355835-856c908f-a025-4a71-885a-d98625e64926.png">

We should add a test for this case and make sure it works.

Reply:
This is by design. I think it's outside the scope to implement word wrapping. Proper word wrapping requires unicode handling and should handle hyphenation. In the context of an update available notification, any custom word wrapping solution would unlikely improve the actual messaging.

Reply:
I think we can get a bit closer by constraining the border width to the column count. I tested this out:

```ts
const maxLine = getMaxWidth(lines, cols);

// ...

/**
 * Find the max width of box border characters we should render based on
 * the width of the content and the terminal columns, whichever is smaller.
 */
function getMaxWidth(lines: [string, number][], terminalColumns: number) {
  const maxFromContent = lines.reduce((p, [, len]) => Math.max(p, len), 0);
  return Math.min(terminalColumns, maxFromContent);
}
```

<img width="317" alt="Screenshot 2023-02-07 at 4 52 07 PM" src="https://user-images.githubusercontent.com/41545/217385454-2c5155d3-8b9a-4b49-8f9b-f0a151eabd1f.png">

Reply:
Gotcha! Fixed!